screens/search_screen/search.tsx

```diff
@@ -1,64 +1,85 @@
-import React, { useState, FC } from 'react';
-import { Text, View, StyleSheet, TextInput, Image, Button, Pressable } from 'react-native'
+import React, { useState, useEffect, FC } from "react";
+import {
+  Text,
+  View,
+  StyleSheet,
+  TextInput,
+  Image,
+  Button,
+  Pressable,
+  Alert,
+  Platform,
+} from "react-native";
+import * as Location from "expo-location";
+import { Ionicons } from "@expo/vector-icons";
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { StackParamList } from "../../App";
 
-import { Ionicons } from '@expo/vector-icons';
+type SearchScreenProps = NativeStackScreenProps<StackParamList, "Search">;
+const Search: FC<SearchScreenProps> = (props) => {
+  const [location, setLocation] = useState("");
+  const [city, setCity] = useState("");
 
+  useEffect(() => {
+    (async () => {
+      let { status } = await Location.requestForegroundPermissionsAsync();
+      if (status !== "granted") {
+        alert("Permission to access location was denied");
+      }
 
-interface Props {
-  city: string;
-  setCity: React.Dispatch<React.SetStateAction<string>>;
-}
-
-const Search: FC<Props> = (props) => {
+      let currentLocation: any = await Location.getCurrentPositionAsync({});
+      setLocation(currentLocation);
+    })();
```

Comment:
Nicely done with the cleanup function.  It took me a lot of head scratching to figure out how to do an async call from within `useEffect`